src/sqlfluff/parser/grammar.py

```diff
@@ -463,24 +463,30 @@ def _bracket_sensitive_look_ahead_match(cls, segments, matchers, parse_context,
             return ((), MatchResult.from_unmatched(segments), None)
 
         # Get hold of the bracket matchers from the dialect, and append them
-        # to the list of matchers.
-        # TODO: Potentially have error handling here for dialects without
-        # square brackets.
-        start_brackets = [
-            parse_context.dialect.ref('StartBracketSegment'),
-            parse_context.dialect.ref('StartSquareBracketSegment')
-        ]
-        end_brackets = [
-            parse_context.dialect.ref('EndBracketSegment'),
-            parse_context.dialect.ref('EndSquareBracketSegment')
-        ]
+        # to the list of matchers. We get them from the relevant set on the
+        # dialect. We use zip twice to "unzip" them. We ignore the first
+        # argument because that's just the name.
+        _, start_bracket_refs, end_bracket_refs, definitely_bracket = zip(*parse_context.dialect.sets('bracket_pairs'))
+        # These are currently strings which need rehydrating
+        start_brackets = [parse_context.dialect.ref(seg_ref) for seg_ref in start_bracket_refs]
+        end_brackets = [parse_context.dialect.ref(seg_ref) for seg_ref in end_bracket_refs]
+        start_definite = list(definitely_bracket)
```

Comment:
What does `definitely_bracket` contain here? I wasn't sure, so `list(definitely_bracket)` confused me. I had guessed it was either `True` or `False`.